app/scripts/lib/util.js

```diff
@@ -158,6 +159,35 @@ function getChainType(chainId) {
   return 'custom';
 }
 
+/**
+ * Loops through the nested allCollectibles collection and finds and returns a collectible
+ * that matches the contractAddress/tokenId pair passed as args.
+ *
+ * @param {Object} allCollectibles - The collection of all collectibles the user has ever owned organized by account and chainId.
+ * @param {string} contractAddress - The address of the contract to match against.
+ * @param {string} collectibleTokenId - The collectible tokenId to match against.
+ * @returns {Object | undefined} - The matching collectible if found, undefined if not found.
+ *
+ */
+function getKnownCollectible(
+  allCollectibles,
+  contractAddress,
+  collectibleTokenId,
+) {
+  let matchedCollectible;
+  find(allCollectibles, (collectiblesByChainId) => {
+    find(collectiblesByChainId, (collectibles) => {
+      matchedCollectible = collectibles.find(
+        ({ address, tokenId }) =>
+          isEqualCaseInsensitive(address, contractAddress) &&
+          tokenId === collectibleTokenId,
+      );
+    });
+  });
+
+  return matchedCollectible;
```

Comment:
Gah I didn't write that correctly 😣  You were right the first time:

``` suggestion
  Object.values(allCollectibles).forEach((collectiblesByChainId) => {
    Object.values(collectiblesByChainId).forEach((collectibles) => {
      const matchedCollectible = collectibles.find(
        ({ address, tokenId }) =>
          isEqualCaseInsensitive(address, contractAddress) &&
          tokenId === collectibleTokenId,
      );
      
      if (matchedCollectible) {
        return matchedCollectible;
      }
    });
  });
```

Reply:
I modified your version so that it should work. Is it better to use `Object.values` over find for the top two loops?

Reply:
actually I've realized this method isn't required at all... 😅 We know the accountAddress and chainId from the transaction data. I've removed and replaced it here: https://github.com/MetaMask/metamask-extension/pull/13048/commits/e512cf0c97640b8f8c9808ee6b7601112d4a2c37 (among a few other cleanups)

Reply:
Additionally I've realized we need to enhance the `checkAndUpdateSingleCollectibleOwnershipStatus` method so that it updates the correct Collectible even if the user has switched accounts or chains between when they initialized and when the transaction was confirmed. I'm doing that here and it should be included in the next controllers version:
https://github.com/MetaMask/controllers/pull/672

Reply:
comment noting this here https://github.com/MetaMask/metamask-extension/pull/13048#discussion_r779868923. 

I don't think this should be considered blocking.